Added Getting Started section in the README (#403)
tokio-rs/tracing: README.md

structured, event-based diagnostic information. `tracing` is maintained by the
Tokio project, but does _not_ require the `tokio` runtime to be used.

## Getting Started

> To get started, configure `Cargo.toml` to include `tracing` and `tracing-subscriber` as dependencies. `tracing` is the core library, while `tracing-subscriber` is the part that implements `Subscriber`, which will record the trace. Take a look at the [Project layout](#Project layout) section.

Reviewer: How about:
```suggestion
To get started, add dependencies on `tracing` and `tracing-subscriber` to your `Cargo.toml`. The `tracing` crate provides the _instrumentation_ APIs for emitting traces, while `tracing-subscriber` provides implementations of the `Subscriber` trait, which will record the trace. For more information on the crates that make up the `tracing` project, take a look at the [Project layout](#Project layout) section.
```

> ```

Reviewer: Add 
```suggestion
```toml
```
to get toml syntax highlighting.

[dependencies]
tracing = "0.1.5"
tracing-subscriber = "0.1.5"
```

> Next, in our `main.rs` file, we will start by adding the libraries

Reviewer: How about:
```suggestion
Next, we will import those crates in our `main.rs` file:
```

> ```

Reviewer: Add 
```suggestion
```rust
```
to get Rust syntax highlighting here.

use tracing::{debug, info, instrument};
use tracing_subscriber::fmt;
```

> For this simple example, we will create a function that prints sequence of numbers and we will instrument that cod to enable tracing. Tracing is enabled by initiating a span and the `#[instrument]` attribute, implicitly create a span for that function.

Reviewer: Typo:
```suggestion
For this simple example, we will create a function that prints sequence of numbers and we will instrument that code to enable tracing. Tracing is enabled by initiating a span and the `#[instrument]` attribute, implicitly create a span for that function.
```

Reviewer: How about
```suggestion
For this simple example, we will create a function that prints sequence of numbers and we will instrument that cod to emit traces. Tracing is enabled by initiating a span and the `#[instrument]` attribute, implicitly create a span for that function.
```

Author: agree 👍

> ```

Reviewer: ```suggestion
```rust
```

// using instrument to setup a span and skip to exclude args
#[instrument(skip(_too_long_string))]

> fn count(nums: i32, _too_long_string: &str) -> Vec<i32> {

Reviewer: What do you think about starting with a basic example of the use of `instrument`, like:
```rust
#[instrument]
fn count(nums: i32) -> Vec<i32>
```
and _then_ introducing the `skip` argument after showing the basic usage first?

Author: I'm, on board with this. I was thinking that I could introduce an argument to the instrument and explain it later down the road. But, yes, let's go with basic usage first.

    let mut seq = vec![];

    for n in 0..=nums {
        // logging n in structured form
        debug!(n, "Next Number");
        seq.push(n);
    }

    seq
}
```

> Then, in our `main()` function, we can create a subscriber, that records the trace. The subscriber employs an environment filter that we can utilize to filter the level of tracing message.

Reviewer: How about:
```suggestion
Once we have added instrumentation to our code, we will need to set up a `Subscriber` to record traces. We'll do this in our `main()` function. The subscriber employs an environment filter that we can utilize to filter the level of tracing message.
```

Reviewer: Also, we should probably link to the `Subscriber` trait's documentation at https://docs.rs/tracing/0.1.10/tracing/trait.Subscriber.html and possibly also to the docs on setting a subscriber at https://docs.rs/tracing/0.1.10/tracing/dispatcher/index.html#using-the-trace-dispatcher

> ```

Reviewer: ```suggestion
```rust
```

fn main() {

>   // this creates subscriber with an environment filter

Reviewer: Personally, I'd prefer to start by introducing how to create and set a subscriber with the default filter
```rust
let subscriber = fmt::Subscriber::default();
tracing::subscriber::with_default(subscriber, || {
    // ...
});
```
and then introduce customizing the filter _after_ we demonstrate the default...

  const DEFAULT_FILTER: &str = concat!(module_path!(), "=", "info");
  let filter = tracing_subscriber::EnvFilter::try_from_default_env()
      .unwrap_or_else(|_| DEFAULT_FILTER.into());
  let subscriber = fmt::Subscriber::builder().with_env_filter(filter).finish();

>   ...

Reviewer: it would be nice if this was valid rust code that actually compiles...we could make the "..." a comment so it's not a syntax error:
```suggestion
  // ...
```

}
```

Finally, we can run the function that prints out the tracing message while it's being run

> ```

Reviewer: ```suggestion
```rust
```

fn main() {

>   ...

Reviewer: again, consider:
```suggestion
  // ...
```

The crates included as part of Tracing are:

> - [`tracing-futures`]: Utilities for instrumenting `futures`.

Reviewer: The change in markdown here seems unrelated to the rest of the change...was this suggested by `markdownlint` or something?

Author: ah, my bad, I think that's my lint doing.. will undo the changes

[`tracing-serde`]: tracing-serde
[`tracing-subscriber`]: tracing-subscriber

> [`tracing-tower`]: tracing-tower

Reviewer: More unrelated changes.
